### ISSUE TITLE
String colors and styling

### ISSUE THREAD

**Issue by nivekcode** [oven-sh/bun#1629]
### What is the problem this feature would solve?

When implementing command line tools I often style strings to make the CLI tool more compelling. To do so I use third-party tools such as `chalk`for example. It seems that currently bun doesn't support any colors or styles such as underlines. Running the following code simply outputs a white string:

```javascript
import chalk from "chalk";

console.log(chalk.underline(chalk.blue('i am blue, duble di duble day')));
```

Thx a lot for this outstanding project, am trying out bun for the first time and its really great ❤️ 

### What is the feature you are proposing to solve the problem?

Support colors and text styles.

**Comment by Electroid** [comment]
Thanks for reporting! Yes, we want packages like `chalk` to just work. We are probably missing a Node.js API or two, preventing it from working.

**Comment by zhatongning** [comment]
Is there a plan to support this?

**Comment by Jarred-Sumner** [comment]
It should work, but I suspect there's either an environment variable missing or we need to more directly check terminfo to determine color support more effectively.

<img width="581" alt="image" src="https://user-images.githubusercontent.com/709451/224463278-43ff0b06-32d9-47f3-8d67-7f4bb3ae2e10.png">

**Comment by Electroid** [comment]
This has been fixed, tested as of Bun v1.0.7.

<img width="213" alt="Screenshot 2023-10-25 at 3 08 58 PM" src="https://github.com/oven-sh/bun/assets/3238291/924df31f-085c-4031-9431-f7bcaa1ae706">
